src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -164,39 +164,51 @@ def is_array_parallelisable(self, loop_variable, var_info):
         for access in var_info.all_accesses:
-            list_of_indices = access.indices
+            indices_groups = access.indices_groups
+
             # Now determine all dimensions that depend
-            # on the parallel variable:
-            for dimension_index, index_expression in \
-                    enumerate(list_of_indices):
-                accesses = VariablesAccessInfo()
-
-                index_expression.reference_accesses(accesses)
-                if Signature(loop_variable) not in accesses:
-                    continue
-
-                # if a previously identified index location does not match
-                # the current index location (e.g. a(i,j), and a(j,i) ), then
-                # the loop can not be parallelised
-                if found_dimension_index > -1 and \
-                        found_dimension_index != dimension_index:
-                    self._add_warning("Variable '{0}' is using loop "
-                                      "variable {1} in index {2} and {3}."
-                                      .format(var_info.var_name,
-                                              loop_variable,
-                                              found_dimension_index,
-                                              dimension_index))
-                    return False
-                found_dimension_index = dimension_index
-                all_indices.append(index_expression)
+            # on the parallel variable. This outer loop is over
+            # the indices used for each component, e.g.
+            # a(i,j)%b(k) it would first handle `(i,j)`, then
+            # `(k)`.
+            for component_index, index_expressions in \
+                    enumerate(indices_groups):
+
+                # This inner loop loop over all indices for the
+                # current component, i.e. `[i, j]` for the first
+                # component above, then `[k]`.
+                for dimension_index, index_expression in \
+                        enumerate(index_expressions):
+                    # Add all the variables used in the index expression
+                    accesses = VariablesAccessInfo()
+                    index_expression.reference_accesses(accesses)
+                    if Signature(loop_variable) not in accesses:
+                        continue
+
+                    # if a previously identified index location does not match
+                    # the current index location (e.g. a(i,j), and a(j,i) ),
+                    # then the loop can not be parallelised
+                    ind_pair = (component_index, dimension_index)
+                    if found_dimension_index[0] > -1 and \
+                            found_dimension_index != ind_pair:
+                        self._add_warning("Variable '{0}' is using loop "
+                                          "variable '{1}' in index '{2}'' "
+                                          "and '{3}'."
```

Comment:
Please could you elaborate this message little as the user is unlikely to understand the meaning of the raw tuples in `found_dimension_index` and `ind_pair`.

Reply:
Postponed to #1268